input/elasticapm/result.go

```diff
@@ -23,18 +23,34 @@ import (
 
 type Result struct {
 	errorsSpace [5]error
+
 	// Errors holds a limited number of errors that occurred while
 	// processing the event stream. If the limit is reached, the
 	// counters above are still incremented.
 	Errors []error
+
 	// Accepted holds the number of valid events accepted.
-	Accepted int
+	Accepted            int
+	SpanAccepted        int
+	TransactionAccepted int
+	MetricAccepted      int
+	LogAccepted         int
```

Comment:
I have a suggestion to make all these counters a bit more digestible:

```go
type Result struct {
	errorsSpace [5]error

	// Errors holds a limited number of errors that occurred while
	// processing the event stream. If the limit is reached, the
	// counters above are still incremented.
	Errors []error

	// Accepted holds the number of valid events accepted.
	Accepted        int
	AcceptedDetails ProcessedDetail

	// TooLarge holds the number of events that were rejected due
	// to exceeding the event size limit.
	TooLarge        int
	TooLargeDetails ProcessedDetail

	// Invalid holds the number of events that were rejected due
	// to being invalid, excluding those that are counted by TooLarge.
	Invalid        int
	InvalidDetails ProcessedDetail
}

// ProcessedDetail holds the number of events processed for each type.
type ProcessedDetail struct {
	Transaction int
	Span        int
	Metric      int
	Log         int
	Error       int
}
```

However, I don't see where the TooLarge / Invalid counters are incremented, did you forget to include the patch?

Reply:
Agree with Mark here, also since it's public type, for all export fields we should add a godoc comment // with a short description.

Reply:
@marclop I'll use the `ProcessedDetail` approach. I was actually thinking of not having a detailed breakdown for  `TooLarge` and `Invalid`, since both error types are basically saying there is an error in the body or in reading the body properly. So encoding the signal type makes no sense, since there might not a signal. For example, from [L214](https://github.com/elastic/apm-data/blob/485554733eceacce688dec287801b1e699c2c7bf/input/elasticapm/processor.go#L214), an Invalid error is captured if no signal type could be deduced.

Reply:
Relating to the issue this PR is solving: We only want to have a detailed breakdown of the _accepted_ events in our dashboards.